dbt/adapters/snowflake/relation.py

```diff
@@ -14,17 +31,18 @@ class SnowflakeRelationType(StrEnum):
     DynamicTable = "dynamic_table"
 
 
-@dataclass
-class SnowflakeQuotePolicy(Policy):
-    database: bool = False
-    schema: bool = False
-    identifier: bool = False
-
-
 @dataclass(frozen=True, eq=False, repr=False)
 class SnowflakeRelation(BaseRelation):
+    # we need to overwrite the type annotation for `type` so that mashumaro reads the class correctly
     type: Optional[SnowflakeRelationType] = None  # type: ignore
+
+    include_policy: SnowflakeIncludePolicy = field(
+        default_factory=lambda: SnowflakeIncludePolicy()
+    )
     quote_policy: SnowflakeQuotePolicy = field(default_factory=lambda: SnowflakeQuotePolicy())
+    relation_configs = {
```

Comment:
Why is this plural if there is only one? We got more coming?

Reply:
Yes, the idea is that if this new approach holds water, then we would eventually migrate all materializations to this approach as needed. Also, some of the methods that use this below would eventually move up into `dbt-core`. This would be the way to "register" your relation config with a relation type, so to speak.